var/spack/repos/builtin/packages/hybrid-lambda/package.py

```diff
@@ -33,9 +33,7 @@ class HybridLambda(AutotoolsPackage):
     # for instance depends_on('boost +filesystem')
     # See https://github.com/spack/spack/pull/22303 for reference
     depends_on(Boost.with_default_variants)
-    depends_on('cppunit', type='test')
-
-    build_directory = 'src'
+    depends_on('cppunit', type=('test', 'link'))
```

Comment:
This will add `cppunit` unconditionally when building `hybrid-lambda`, so I don't think it's correct. What is linking to the library? Hopefully it's just unit-tests, not the library that should be consumed by users.

Reply:
Thank you. This bug is that cppunit lib can not be found at link phase when use '--test' option.  In spack manual, it seems that there is no exactly depend type such as 'test & link'. If there is, please tell me.

----------
“build”: the dependency will be added to the PATH and PYTHONPATH at build-time.

“link”: the dependency will be added to Spack’s compiler wrappers, automatically injecting the appropriate linker flags, including -I, -L, and RPATH/RUNPATH handling.

“run”: the dependency will be added to the PATH and PYTHONPATH at run-time. This is true for both spack load and the module files Spack writes.

“test”: the dependency will be added to the PATH and PYTHONPATH at build-time. The only difference between “build” and “test” is that test dependencies are only built if the user requests unit tests with spack install --test.

Reply:
Hmm, I need to double check this, but at a first glance it seems a lack of the build-system or the recipe. Testing a package shouldn't result in the test dependencies to be linked in the installed artifacts.

Reply:
I've found a method that it can resolve this problem. @alalazo 
The depends_on type of "cppunit" does not need to be changed. Just append cppunit lib path to LD_LIBRARY_PATH in setup_build_environment method when run_tests is true.